### DISCUSSION TITLE
Is Zed only useable on GPU-enabled machines?

### DISCUSSION THREAD

**Discussion by thor314** [zed-industries/zed#17212] [Feature Requests]
It seems that Zed complains my linux laptop does not have an onboard GPU:
> zed uses vulkan for rendering and requires a compatible gpu. you are using a software emulated gpu, which will result in awful performance.

 From that message, I infer my options are:
 - get a beefier laptop with a GPU
 - don't use Zed
 
 is that correct?

**Comment by bajrangCoder** [comment]
No, It's works well on integrated GPU.
I have an old Intel processor with no dedicated GPU but Zed works flawlessly and it's fast . I'm using Arch linux

**Reply by thor314** [reply]
how do i set that up?

**Reply by bajrangCoder** [reply]
Install the vulkan graphics lib and then install zed using any of mentioned method from zed docs

**Reply by thor314** [reply]
Maybe I misunderstand what you mean.


I've tried:
```
sudo apt install vulkan-tools libvulkan1 libvulkan-dev
sudo apt install mesa-vulkan-drivers
vulkaninfo # returns a long text block of info

# opens zed with the warning, Unsupported GPU
zed
```

edit: 
I believe I misunderstood earlier what an integrated GPU is. I don't have one of those, so I don't believe I'll be able to run zed. Cheers, thanks for the comments.

**Reply by gkobeaga** [reply]
I successfully ran the Zed editor on Arch Linux with an integrated Intel GPU by installing the `vulkan-intel` package.

**Reply by neurorune** [reply]
I also have an integrated gpu intel irish xe and my laptop runs hot when I use zed. The gpu is constantly at 50-69% utilization on an empty open folder. So when I have webpages and youtube open it gets to 100%. Not good for laptop. Even VSCode doesn't use that much gpu only at 1%-5%. Zed is a performative editor. It's like throwing money at it just keep it running

**Reply by driehuis** [reply]
FWIW, my installation of zed also complains about the GPU. I've got an older Kaby Lake system with integrated GPU, which glxgears recognises and uses to great effect. But for some reason, zed doesn't pick it up. `vulkaninfo` shows, in tons of other info, this info about my GPU's:
```
VK_LAYER_INTEL_nullhw (INTEL NULL HW) Vulkan version 1.1.73, layer version 1:
	Layer Extensions: count = 0
	Devices: count = 2
		GPU id = 0 (Intel(R) Iris(R) Plus Graphics 650 (Kaby Lake GT3e) (KBL GT3))
		Layer-Device Extensions: count = 0

		GPU id = 1 (llvmpipe (LLVM 15.0.7, 256 bits))
		Layer-Device Extensions: count = 0
```

I'm totally out of my depth with Vulkan and its Rust interface, but seeing that video without any hardware GPU support is rare these days even on older hardware, this may help someone formulate a more specific issue for Zed to allow fixing what may be the underlying issue.

**Comment by muhiris** [comment]
Zed, you just broke old machine compatibility by mandating Vulkan support. You locked out users with older hardware or systems lacking proper GPU drivers. Your decision to force Vulkan rendering without maintaining a fallback option demonstrates poor judgment regarding hardware accessibility.

This is exactly how you kill adoption and fragment the user base. Not everyone runs cutting-edge hardware, and many developers work on older but perfectly functional machines. Watch as someone forks this and maintains a version that actually respects hardware diversity.

Terrible product decision. Back to VS Code.

**Reply by obamarama** [reply]
totally agree with @muhiris! i'm an active python developer and have a couple of computers for my work that performs great enough with minimalistic arch setup, but my GPU's just aren't new enough to use vulkan, so i couldn't even try Zed while really interested in it.

**Reply by zakutin** [reply]
Indeed, I have an old ThinkPad X230 with an Intel HD 4000 integrated graphics card that runs Fedora perfectly. However, the performance is terrible: there is a high GPU load even when I simply move the mouse over an empty editor and the cooler spins like crazy.

<img width="2560" height="1440" alt="Screenshot From 2025-07-31 11-41-30" src="https://github.com/user-attachments/assets/78c83044-3077-4cfd-a938-78ccbd126623" />

**Comment by NesBet** [comment]
For those without beefier or dedicated GPUs here is a solution I used on my old machine:
1. Install vulkan-swrast (I used yay -S vulkan-swrast).
2. Then install zed using any method on their website
3.  Launch zed and enjoy!

**Reply by muhiris** [reply]
This is the command to install on Debian systems.
`sudo apt install mesa-vulkan-drivers mesa-vulkan-drivers:i386`

ps: Doesn't Work for me. Uninstalled Zed and It's dependencies
```
sudo apt remove vulkan-tools libvulkan1 libvulkan-dev
sudo apt remove mesa-vulkan-drivers mesa-vulkan-drivers:i386
sudo apt remove zed
sudo apt autoremove
```

**Comment by rwalle** [comment]
I tried using Zed on a Linux virtual machine environment provided by work where there is no GPU. Almost everything else works -- chrome, firefox, vscode, vnc etc, you name it. You can even watch Youtube videos on it if you want. But no Zed.

I guess I'll go back to vscode then.

**Comment by reeshabh90** [comment]
I have Intel HD Graphics 3000 integrated into a 3rd Generation Core processor. No vulkan support it seems. Zed just loads the system. I mean what is the point of developing an IDE which can not even support basic hardware setup.

**Comment by driehuis** [comment]
For all those commenting: have you tried the debugging steps in the documentation? There are many small bugs that can cause zed to fail for which workarounds exist.

In my case, `MESA_VK_DEVICE_SELECT=list zed --foreground` showed that blade_graphics misidentified my system as having an Nvidia graphics card. In issue #22978, I found a workaround for my case: `sudo rm /etc/prime-discrete` (heaven knows what put that file there, five years ago).

Vulkan supports many older Intel integrated GPU's. If your PC or laptop is less than ten years old but fails to run zed, chances are that your hitting a bug.

**Reply by muhiris** [reply]
That misses the bigger point. Zed used to work on older hardware, and now it doesn’t unless specific GPU requirements are met. When a project regresses like this, it’s fair to question the decision itself, not just hunt for workarounds. This feels intentional, not accidental.

**Reply by reeshabh90** [reply]
> For all those commenting: have you tried the debugging steps in the documentation? There are many small bugs that can cause zed to fail for which workarounds exist.
> 
> In my case, `MESA_VK_DEVICE_SELECT=list zed --foreground` showed that blade_graphics misidentified my system as having an Nvidia graphics card. In issue #22978, I found a workaround for my case: `sudo rm /etc/prime-discrete` (heaven knows what put that file there, five years ago).
> 
> Vulkan supports many older Intel integrated GPU's. If your PC or laptop is less than ten years old but fails to run zed, chances are that your hitting a bug.

There are no errors coming for now. Zed got installed on freeBSD. But its performance is just worse. And it itself displays the message about performance on vulkan support.

**Reply by reeshabh90** [reply]
> That misses the bigger point. Zed used to work on older hardware, and now it doesn’t unless specific GPU requirements are met. When a project regresses like this, it’s fair to question the decision itself, not just hunt for workarounds. This feels intentional, not accidental.

I was looking at Zed as an alternative, since in VS code, one can not disable telemetry. Zed being open-source looked promising.

**Reply by JulienGAMartin** [reply]
I tried the debugging steps. Computer is 3 years old and work flawlessly for everything that do not require specific GPU. A text editor should not have specific GPU requirement that drastically limit performance on a recent computer.

**Comment by JulienGAMartin** [comment]
Well I m highly disappointed by Zed requirement of a GPU. You cannot gloat about performance when you only require high perfromance hardware. Everything works flawlessly under VS Code (and derivative) on my computer.
This is Zed warning message "Currently you are using a software emulated GPU (llvmpipe (LLVM 19.1.1, 256 bits)) which will result in awful performance"

Based on the message and the troubleshooting page, there is not even a point in trying to use it.  Zed is crap on my computer.  

However Zed is working after the warning and seems to work OK (read the performance is far from awful). So I would strongly suggest to rephrase the warning message and the troubleshooting page and indicate what performance decrease is expected instead of motivating people to either not use Zed or buy a GPU.

This might be the end of my journey with Zed.

**Comment by andreujuanc** [comment]
I was looking for alternatives as every instance of vscode was eating like 7GB of ram. 
Zed barely uses any of it, but it's super slow. 
I'm inside an ARM linux on Parallels. 
Everything works fine, youtube 4k, chrome, vscode, firefox, games with arm support, you name it, zed runs but feels sluggish :/

**Comment by wbulot** [comment]
Just wanted to add my disappointment here too.
I mainly use IDEs / Ai agent on virtual machines without GPU support, and Zed is the only tool that cannot work on them.
That GPU requirement is ridiculous.

**Reply by silver-dragon** [reply]
I am in the same boat. My workflow necessitates the use of VMs which often don't have GPU support. So I am unable to launch Zed.

**Reply by wbulot** [reply]
I switched back to VSCode, using Roo Code for agentic coding. Best compromise for me.

**Comment by vizvayu** [comment]
I need to use Zed in virtual machines without GPU support as well, and it's painfully slow, but the worst part is the high CPU usage, even in idle.

**Comment by miranquil** [comment]
I'm using Citrix VDI in my working space. Zed's behavior is so bad that I cannot use it with any efficiency.

**Comment by reecelikesramen** [comment]
I'm using Zed on an Azure Windows 11 VM for work with no GPU. The performance is awful. I love Zed on laptops with integrated GPUs fine, but not having a good software fallback is a big problem.

**Comment by silver-dragon** [comment]
I am using [Cendio ThinLinc ](https://www.cendio.com/) with [virtualGL](https://virtualgl.org/)

`vglrun zed` results in a window launching but with no contents nothing renders. 

Have gpu and drivers installed other software works fine with vglrun and offers GPU acceleration.

**Comment by garcipat** [comment]
I'm developing on a VM. I don't have GPU support there, and it shouldn't be necessary.

**Comment by sininspira2** [comment]
Also in a situation where I have developers using VMs without GPU-backing. Very disappointing, Zed.

**Comment by z3r0n3br4instorm** [comment]
Hi everyone, I am maintaining this fork of zed with OpenGL3.3 for my MacBook Pro 2011 running linux. You can build it using this if anyone is interested. 

https://github.com/z3r0n3br4instorm/zed/tree/gl33-backend 

Here's the pre-compiled binary: https://drive.google.com/file/d/1BwfKQZ244XbmhXzCnXNBHm9PPlfuZTIr/view?usp=drive_link

**Comment by freekvh** [comment]
Yeah, VMs are proposed as a way to have your agents get some more freedom in our company, but then we get this error message. Not sure how serious I should take it, I'm just going to see how it goes. I use Gnome Boxes and enabled 3D support, still get the warning though.
